### PR TITLE
AP-2455-2476-2506 User Management refinements

### PR DESCRIPTION
**Pull request by dreamind**

### CODE REVIEW

**Review by raboczi — COMMENTED** [review]
I would prefer ComponentUtil not be in Apromore-Commons because it introduces a dependency on ZK.  Other services that depend on Apromore-Commons such as Apromore-Calendar shouldn't require the ZK libraries.

File: Apromore-Commons/pom.xml [hunk 1]
```diff
@@ -66,6 +66,12 @@
 			<version>1.4.2</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.zkoss.zk</groupId>
+			<artifactId>zul</artifactId>
+			<version>${zk.version}</version>
+			<optional>true</optional>
+		</dependency>
```
Comment (nolantellis) [inline Apromore-Commons/pom.xml]: I agree with Simon. This should be in some Generic UI module. For now Portal sounds right for a quick move.

**Comment by dreamind** [comment]
ComponentUtils is now moved under portal

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/zk/ComponentUtils.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.slf4j.LoggerFactory;
+
+public final class ComponentUtils {
+    private static final String SCLASS_OFF = "ap-state-off";
+    private static final String SCLASS_ON = "ap-state-on";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ComponentUtils.class.getName());
+
+    public static void toggleSclass(HtmlBasedComponent comp, boolean newState, String sclassOff, String sclassOn) {
+        String sclass = comp.getSclass();
```
Comment (nolantellis) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/zk/ComponentUtils.java:36]: Not important now, But StringBuilder will be great in this place.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/zk/ComponentUtils.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        comp.setSclass(sclass);
+    }
+
+    public static void toggleSclass(HtmlBasedComponent comp, String sclassOff, String sclassOn) {
+        String sclass = comp.getSclass();
+        if (sclass == null) {
+            sclass = "";
+        }
+        if (sclass.contains(sclassOn)) {
```
Comment (nolantellis) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/zk/ComponentUtils.java:55]: pass sclass.contains(sclassOn) as argument to the method and remove if else block

**Review by nolantellis — APPROVED** [review]

**Pull request merged**